.github/workflows/ruby-core-test-suite.yml

```diff
@@ -0,0 +1,52 @@
+name: ruby-core-test
+
+on:
+  pull_request:
+
+  push:
+    branches:
+      - master
+
+jobs:
+  ruby_core_test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: [rubygems, bundler]
+    steps:
+      - name: Install libraries
+        run: |
+          set -x
+          sudo apt-get update -q || :
+          sudo apt-get install --no-install-recommends -q -y build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm-dev bison autoconf ruby
+      - name: Prepare the working directory
+        run: |
+          mkdir ruby
+          mkdir rubygems
+      - uses: actions/checkout@v2
+        with:
+          repository: "ruby/ruby"
+          path: ruby/ruby
+      - name: Build Ruby
+        run: |
+          autoconf
+          ./configure -C --disable-install-doc
+          make -j2
+        working-directory: ruby/ruby
+      - uses: actions/checkout@v2
+        with:
+          path: rubygems/rubygems
+      - name: Test RubyGems
+        run: |
+          ruby tool/sync_default_gems.rb rubygems
+          make test-all TESTS="rubygems -j2"
+        working-directory: ruby/ruby
+        continue-on-error: true
```

Comment:
I see that the check is actually broken at the moment (I seem to have broken it again when moving our setup to use bundler to manage development dependencies?).

So I guess the idea is to start with this as an allowed failure, and once we fix the current situation, we start enforcing it to be green, right?

That approach sounds good to me! :+1:

Reply:
Yes, The test of current master branch with ruby master failed now. I will work after merging this.